fastly/data_source_fastly_tls_certificate_ids.go

```diff
@@ -0,0 +1,46 @@
+package fastly
+
+import (
+	"fmt"
+	"github.com/hashicorp/terraform-plugin-sdk/helper/hashcode"
+	"github.com/hashicorp/terraform-plugin-sdk/helper/schema"
+)
+
+func dataSourceFastlyTLSCertificateIDs() *schema.Resource {
+	return &schema.Resource{
+		Read: dataSourceFastlyTLSCertificateIDsRead,
+		Schema: map[string]*schema.Schema{
+			"ids": {
+				Type:        schema.TypeList,
+				Description: "List of IDs corresponding to Custom TLS certificates.",
+				Computed:    true,
+				Elem:        &schema.Schema{Type: schema.TypeString},
+			},
+		},
+	}
+}
+
+func dataSourceFastlyTLSCertificateIDsRead(d *schema.ResourceData, meta interface{}) error {
+	conn := meta.(*FastlyClient).conn
+
+	certificates, err := listTLSCertificates(conn)
+	if err != nil {
+		return err
+	}
+
+	var ids []string
+	for _, certificate := range certificates {
+		ids = append(ids, certificate.ID)
+	}
+
+	// 2.x upgrade note - `hashcode.String` was removed from the SDK
+	// Code will need to be copied into this repository
+	// https://www.terraform.io/docs/extend/guides/v2-upgrade-guide.html#removal-of-helper-hashcode-package
+	d.SetId(fmt.Sprintf("%d", hashcode.String(""))) // hashCode should include any filters set when they are added
```

Comment:
Maybe the wording isn't clear - it's supposed to indicate that, if additional filters are added to the data source (such as `tls_activation_ids`) then the filters should be used instead of the empty string. I'll try and clear this up

Reply:
Awesome, thanks!